### PR TITLE
Bump `:gun` to `2.0.0-rc.2`.

### PR DESCRIPTION
**Pull request by jchristgit**
This includes the new `:gun_pool` module and an upgrade to `cowlib`, both of
which we do not use at all.

### CODE REVIEW

**Comment by Kraigie** [comment]
LGTM if this isn't blocked by the other gun PR

**Review by Kraigie — APPROVED** [review]

**Comment by jchristgit** [comment]
No

**Comment by jchristgit** [comment]
Thanks!

**Pull request merged**
